astropy/units/quantity_helper/function_helpers.py

```diff
@@ -1042,3 +1043,26 @@ def eig(a, *args, **kwargs):
     from astropy.units import dimensionless_unscaled
 
     return (a.value,)+args, kwargs, (a.unit, dimensionless_unscaled), None
+
+
+@function_helper(module=np.lib.recfunctions)
+def structured_to_unstructured(arr, *args, **kwargs):
+    """
+    Convert a structured quantity to an unstructured one.
+    This only works if all the units are compatible.
+    """
+    units = arr.unit.values()
+    unit = units[0]
+
+    # all units are equal
+    if all(np.equal(units, units[0])):
```

Comment:
Just a fly-by comment for now: I think I'd write this more simply by creating a new structure unit which has `units[0]` in all its fields, and then doing `.to_value`, i.e., something along the lines of this trial:
```
In [15]: a = [(5, (4., 3.))] * u.Unit('m, (cm, um)')

In [16]: def replace_unit(x):
    ...:     if isinstance(x, u.StructuredUnit):
    ...:         return x._recursively_apply(replace_unit)
    ...:     else:
    ...:         return u.m
    ...: 

In [17]: a.unit._recursively_apply(replace_unit)
Out[17]: Unit("(m, (m, m))")

In [18]: a.to(a.unit._recursively_apply(replace_unit))
Out[18]: <Quantity [(5., (0.04, 3.e-06))] (m, (m, m))>

In [19]: a.to_value(a.unit._recursively_apply(replace_unit))
Out[19]: 
array([(5., (0.04, 3.e-06))],
      dtype=[('f0', '<f8'), ('f1', [('f0', '<f8'), ('f1', '<f8')])])

```

Reply:
Thanks! That's a much cleaner method to transform the unit.